File: StackMethods/stack.pop/pop.py
class Stack:
    def __init__(self, capacity=8):
        self.stack = []
        self.capacity = capacity

    def is_empty(self):
        return len(self.stack) == 0

    def is_full(self):
        return len(self.stack) == self.capacity

    def push(self, value):
        if self.is_full():
            print(f"Stack Overflow! Cannot insert {value}")
        else:
            self.stack.append(value)

    def pop(self):
        if self.is_empty():
            print("Stack Underflow!")
            return None
        return self.stack.pop()

    def peek(self):
        if self.is_empty():
            print("Stack is empty!")
            return None
        return self.stack[-1]

    def print_stack(self):
        print(" ".join(map(str, self.stack)))


# Test
s = Stack(8)

s.push(44)
s.push(10)
s.push(62)
s.push(123)
s.push(15)

print("Stack Elements:", end=" ")
s.print_stack()

print("Top element:", s.peek())

print("Elements popped:", end=" ")
while not s.is_empty():
    print(s.pop(), end=" ")
print()

File: StackMethods/stack.pop/pop.cpp
#include <iostream>
using namespace std;

class Stack {
private:
    int top;
    int size;
    int* arr;

public:
    // Constructor
    Stack(int capacity) {
        size = capacity;
        arr = new int[size];
        top = -1;
    }

    // Destructor
    ~Stack() {
        delete[] arr;
    }

    bool isEmpty() {
        return top == -1;
    }

    bool isFull() {
        return top == size - 1;
    }

    void push(int data) {
        if (isFull()) {
            cout << "Stack is full. Cannot push " << data << endl;
        } else {
            arr[++top] = data;
            cout << "Pushed: " << data << endl;
        }
    }

    int pop() {
        if (isEmpty()) {
            cout << "Stack is empty. Cannot pop." << endl;
            return -1;
        } else {
            int val = arr[top--];
            cout << "Popped: " << val << endl;
            return val;
        }
    }

    void display() {
        if (isEmpty()) {
            cout << "Stack is empty." << endl;
        } else {
            cout << "Stack elements: ";
            for (int i = 0; i <= top; i++) {
                cout << arr[i] << " ";
            }
            cout << endl;
        }
    }
};

int main() {
    Stack s(5);

    s.push(10);
    s.push(20);
    s.push(30);
    s.push(40);

    s.display();

    // Demonstrate popping
    s.pop();
    s.pop();

    s.display();

    // Optionally, pop remaining elements to show empty stack
    while (!s.isEmpty()) {
        s.pop();
    }
    return 0;
}